qsc/tests/test_to_vmec.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+
+from qsc.to_vmec import to_Fourier
+import unittest
+import os
+from scipy.io import netcdf
+import numpy as np
+import logging
+from qsc.qsc import Qsc
+from mpi4py import MPI
+import vmec
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+def compare_to_fortran(name, filename):
+    """
+    Compare output from pyQSC to the fortran code, for one
+    of the example configurations from the papers.
+    """
+    # Add the directory of this file to the specified filename:
+    abs_filename = os.path.join(os.path.dirname(__file__), filename)
+    f = netcdf.netcdf_file(abs_filename, 'r')
+    nphi = f.variables['N_phi'][()]
+    r = f.variables['r'][()]
+    mpol = f.variables['mpol'][()]
+    ntor = f.variables['ntor'][()]
+
+    logger.info('Creating pyQSC configuration')
+    py = Qsc.from_paper(name, nphi=nphi)
+    logger.info('Outputing to VMEC')
+    py.to_vmec(str("input."+name).replace(" ",""), r=r,params={'mpol': mpol, 'ntor': ntor})
+
+    logger.info('Comparing to fortran file ' + abs_filename)
+    def compare_field(fortran_name, py_field, rtol=1e-9, atol=1e-9):
+        fortran_field = f.variables[fortran_name][()]
+        logger.info('max difference in {}: {}'.format(fortran_name, np.max(np.abs(fortran_field - py_field))))
+        np.testing.assert_allclose(fortran_field, py_field, rtol=rtol, atol=atol)
+
+    compare_field('RBC', py.RBC)
+    compare_field('RBS', py.RBS)
+    compare_field('ZBC', py.ZBC)
+    compare_field('ZBS', py.ZBS)
+    f.close()
+
+def compare_to_vmec(name, r=0.01, nphi=101):
+    """
+    Check that VMEC can run the input file outputed by pyQSC
+    and check that the resulting VMEC output file has
+    the expected parameters
+    """
+    # Add the directory of this file to the specified filename:
+    inputFile="input."+str(name).replace(" ","")
+    abs_filename = os.path.join(os.path.dirname(__file__), inputFile)
+    # Run pyQsc and create a VMEC input file
+    logger.info('Creating pyQSC configuration')
+    py = Qsc.from_paper(name, nphi=nphi)
+    logger.info('Outputing to VMEC')
+    py.to_vmec(inputFile,r)
+    # Run VMEC
+    fcomm = MPI.COMM_WORLD.py2f()
+    logger.info("Calling runvmec. comm={}".format(fcomm))
+    vmec.runvmec(np.array([15,0,0,0,0], dtype=np.int32), inputFile, True, fcomm, '')
+    # Open VMEC output file
```

Comment:
Let's add an assertion here that `ictrl[1] == 11`, which means that VMEC converged. You'll need to first define `ictrl=np.array([15,0,0,0,0], dtype=np.int32)` rather than creating ictrl in the call to runvmec as now.